aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/events/ModelSyncedEvent.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine.events;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.datastore.syncengine.SyncType;
+
+/**
+ * Hub event payload emitted when the initial sync completes for a given model.
+ */
+public final class ModelSyncedEvent {
+    private final int added;
+    private final int updated;
+    private final int deleted;
+    private final String model;
+    private final boolean isFullSync;
+    private final boolean isDeltaSync;
+
+    /**
+     * Constructs a ModelSyncEvent object.
+     * @param model The of the model.
+     * @param syncType Indicates whether is a full or delta sync.
+     * @param added Number of records added during the sync attempt.
+     * @param updated Number of records updated during the sync attempt.
+     * @param deleted Number of records deleted during the sync attempt.
+     */
+    public ModelSyncedEvent(String model,
+                            SyncType syncType,
+                            int added,
+                            int updated,
+                            int deleted) {
+        this.added = added;
+        this.updated = updated;
+        this.deleted = deleted;
+        this.model = model;
+        this.isFullSync = SyncType.FULL.equals(syncType);
+        this.isDeltaSync = SyncType.DELTA.equals(syncType);
+    }
+
+    /**
+     * Getter for the model name.
+     * @return The model name (ex. Post).
+     */
+    public String getModel() {
+        return model;
+    }
+
+    /**
+     * Getter for the number of records added during the sync.
+     * @return Number of records added.
+     */
+    public int getAdded() {
+        return added;
+    }
+
+    /**
+     * Getter for the number of records updated during the sync.
+     * @return Number of records updated.
+     */
+    public int getUpdated() {
+        return updated;
+    }
+
+    /**
+     * Getter for the number of records deleted during the sync.
+     * @return Number of records deleted.
+     */
+    public int getDeleted() {
+        return deleted;
+    }
+
+    /**
+     * Getter for the isFullSync property.
+     * @return True if it's a full sync, false otherwise.
+     */
+    public boolean isFullSync() {
```

Comment:
Or, alternately, you could just use your enum type? `getSyncType()`? The way you have it is fine too.

Reply:
The event payload calls for isFullSync and isDeltaSync, so I just kept it consistent with what JS is doing.